brickstrap.sh

```diff
@@ -19,29 +20,51 @@
 # http://twitter.com/kvz
 #
 
-### Configuration
+### Runtime
 #####################################################################
+set -e
+
+# Bash will remember & return the highest exitcode in a chain of pipes.
+set -o pipefail
 
 # Environment variables
 [ -z "${LOG_LEVEL}" ] && LOG_LEVEL="3" # 4 = debug -> 0 = fail
 
-# Commandline options. This defines the usage page
+#
+# Get the brickstrap source directory
+#
+function br_script_path()
+{
+    SCRIPT_PATH=$(readlink -f "$0")
+    SCRIPT_PATH=$(dirname "$SCRIPT_PATH")
+    echo -n "$SCRIPT_PATH"
+}
 
-read -r -d '' usage <<-'EOF'
-  usage: brickstrap [-b <board>] [-d <dir>] <command>
+. "$(br_script_path)/brickstrap-components.sh"
+
+# Commandline options. This defines the usage page
+BRP_USAGE=$(cat <<'EOF'
+Usage: brickstrap -p <project> [-c <component>] [-d <dir>] <command>
 
 Options
 -------
--b <board> Directory that contains board definition. (Default: ev3dev-jessie)
--d <dir>   The name of the directory for the rootfs. Note: This is also
-           used for the .tar and .img file names.
--f         Force overwriting existing files/directories.
--h         Help. (You are looking at it.)
+-b <board>     Directory that contains board definition. Alias for -p.
+-c <component> Select components from the board directory. Multiple components
+               may be selected by specifying multiple '-c' options, at least
+               one component is required.
+-d <dir>       The name of the directory for the rootfs.
```

Comment:
Actually, I take that back. It think we should leave it as `-d <dir>`. Currently, we create all files in the current directory and the actual rootfs has the same name as `<dir>`. However, I think it would be better if we contained all of the generated files in a parent directory. So, we have...


    <dir>/                        created by create-conf
      + rootfs/                   created by create-rootfs
      + brickstrap.conf           created by create-conf
      + multistrap.conf           created by create-conf
      + rootfs.tar                created by create-tar
      + <dir>.img                 created by create-image
      + <dir>-report.txt          created by create-report